vcd/resource_vcd_vm_sizing_policy.go

```diff
@@ -0,0 +1,604 @@
+							Optional:     true,
+							Description:  "Defines the memory limit in MB for a VM. If not defined in the VM sizing policy, memory limit is equal to the allocated memory for the VM.",
+							ValidateFunc: IsIntAndAtLeast(0),
+						},
+						"shares": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Description:  "Defines the number of memory shares for a VM. Shares specify the relative importance of a VM within a virtual data center. If a VM has twice as many shares of memory as another VM, it is entitled to consume twice as much memory when these two virtual machines are competing for resources. If not defined in the VDC compute policy, normal shares are applied to the VM.",
+							ValidateFunc: IsIntAndAtLeast(0),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceVmSizingPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	policyName := d.Get("name").(string)
+	log.Printf("[TRACE] VM sizing policy creation initiated: %s", policyName)
+
+	vcdClient := meta.(*VCDClient)
+
+	if !vcdClient.Client.IsSysAdmin {
+		return fmt.Errorf("functionality requires system administrator privileges")
+	}
+
+	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrg, err)
+	}
+
+	params, err := getVmZingingPolicyInput(d, vcdClient)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Creating VM sizing policy: %#v", params)
+
+	createdVmSizingPolicy, err := adminOrg.CreateVdcComputePolicy(params)
+	if err != nil {
+		log.Printf("[DEBUG] Error VM sizing policy: %s", err)
+		return fmt.Errorf("error VM sizing policy: %s", err)
+	}
+
+	d.SetId(createdVmSizingPolicy.VdcComputePolicy.ID)
+	log.Printf("[TRACE] VM sizing policy created: %#v", createdVmSizingPolicy.VdcComputePolicy)
+
+	return resourceVmSizingPolicyRead(d, meta)
+}
+
+// resourceVcdVmAffinityRuleRead reads a resource VM affinity rule
+func resourceVmSizingPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	return genericVcdVmSizingPolicyRead(d, meta)
+}
+
+// Fetches information about an existing VM sizing policy for a data definition
+func genericVcdVmSizingPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	policyName := d.Get("name").(string)
+	log.Printf("[TRACE] VM sizing policy read initiated: %s", policyName)
+
+	vcdClient := meta.(*VCDClient)
+
+	org, _, err := vcdClient.GetOrgAndVdc(d.Get("org").(string), "")
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrg, err)
+	}
+
+	// The method variable stores the information about how we found the rule, for logging purposes
+	method := "id"
+
+	var policy *govcd.VdcComputePolicy
+	if d.Id() != "" {
+		policy, err = org.GetVdcComputePolicyById(d.Id())
+		if err != nil {
+			log.Printf("[DEBUG] Unable to find VM sizing policy %s. Removing from tfstate.", policyName)
+			d.SetId("")
+			return fmt.Errorf("unable to find VM sizing policy %s, err: %s. Removing from tfstate", policyName, err)
```

Comment:
Nit. I'm not very keen in ` err: %s.` part, because the error may be of multiple sentences and contain dots, so the "Removing from tfstate" sentence may be read as if part of the error. However, I don't have a better suggestion how to format it more distinctly. Maybe you do?

Reply:
made fast scan for best practises. Seems that everywhere  `:` used.